benthomasson/ansible-events-ui
# AAP-[5705] Project endpoints update description and name

### AAP-5705
Jira ticket: [AAP-5705](https://issues.redhat.com/browse/AAP-5705)

Small updates for project endpoints:
- create endpoint returns a 201 status and only allows unique name parameter
- description parameter added to update method (both name and description are optional)

#### Testing instructions:
1. Create Project: POST `http://localhost:8080/api/projects/`
```
{
    "url": "https://github.com/benthomasson/eda-project",
    "name": "testing",
    "description": "testing"
}
```
Should receive a 201 status and response of the project object

2. Create Project: POST `http://localhost:8080/api/projects/` with the same body should get a 422 response with the following body
```
{
    "detail": "Project with name 'testing' already exists"
}
```
3.  Update Project: PATCH `http://localhost:8080/api/projects/{id}` (should be able to pass full body or 1 parameter)
```
{
    "name": "update test",
    "description": "update testing"
}
```
